app/schemas/activity.py

```diff
@@ -34,6 +36,7 @@ class ActivityCreate(ActivityBase, AuthorizationOptionalPublicMixin):
     generated_ids: list[uuid.UUID] = []
 
 
-class ActivityUpdate(ActivityBase, AuthorizationOptionalPublicMixin):
-    end_time: datetime | None = None
-    generated_ids: list[uuid.UUID] | None = None
+class ActivityUpdate(BaseModel):
+    start_time: datetime | Literal["<NOT_SET>"] | None = NOT_SET
```

Comment:
```suggestion
    start_time: datetime | Literal[NOT_SET] | None = NOT_SET
```

Might be worth making a type for it if it'll be used elsewhere.

Reply:
weird, I checked it w/ python 3.11:
```
>>> NOT_SET = "<NOT_SET>"
>>> from typing import Literal
>>> NST = Literal[NOT_SET]
>>> NST
typing.Literal['<NOT_SET>']

```
which is why I thought it would work :)

Seems to work under 3.12 as well:
```
>>> NOT_SET = "<NOT_SET>"
>>> from typing import Any, Literal
>>> NST = Literal[NOT_SET]
>>> NST
typing.Literal['<NOT_SET>']
>>> import sys; print(sys.version_info)
sys.version_info(major=3, minor=12, micro=8, releaselevel='final', serial=0)
```

Strange.